internal/topo/node/source_node.go

```diff
@@ -140,6 +146,8 @@ func (m *SourceNode) Open(ctx api.StreamContext, errCh chan<- error) {
 						for {
 							select {
 							case <-ctx.Done():
+								// schema may be changed after we close the topo
```

Comment:
The sourceNode shall be destroyed after closing. Find a problem of closing?

Reply:
I add the reason in comment